app/src/main/java/com/example/dinhh/soundscape/presentation/screens/library/LibraryFragment.kt

```diff
@@ -1,27 +1,71 @@
 package com.example.dinhh.soundscape.presentation.screens.library
 
-
+import android.arch.lifecycle.Observer
 import android.os.Bundle
 import android.support.v4.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.ListView
+import android.widget.Toast
 import com.example.dinhh.soundscape.R
+import com.example.dinhh.soundscape.data.Model
+import com.example.dinhh.soundscape.presentation.ListAdapter
+import com.example.dinhh.soundscape.presentation.ListItem
+import com.example.dinhh.soundscape.presentation.screens.sounds.SoundsFragment
+import org.koin.android.viewmodel.ext.android.viewModel
 
 class LibraryFragment : Fragment() {
 
+    private val libraryViewModel: LibraryViewModel by viewModel()
+    private var isClicked = false
+
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        // Inflate the layout for this fragment
         return inflater.inflate(R.layout.fragment_library, container, false)
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val listView = view.findViewById<ListView>(R.id.listView)
+        listView.adapter = ListAdapter(context!!, Model.category)
+        listView.setOnItemClickListener{ _, _, position, _ ->
+            isClicked = true
+            val selectedCategory = listView.getItemAtPosition(position) as ListItem
+            libraryViewModel.beginSearch(selectedCategory.toString())
+        }
+
+        libraryViewModel.viewState.observe(this, Observer {
+            it?.run(this@LibraryFragment::handleView)
+        })
+    }
+
+    private fun handleView(viewState: LibraryViewState) = when (viewState) {
+        is LibraryViewState.Success -> {
+            if (isClicked) {
+                goToSoundsFragment()
+                isClicked = false
+            }else{
+            }
+        }
+        is LibraryViewState.Failure -> {
+            Toast.makeText(activity, "Error: ${viewState.throwable.localizedMessage}", Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    fun goToSoundsFragment(){
+        val fragManager = fragmentManager
+        val fragmentTransaction = fragManager?.beginTransaction()
+        fragmentTransaction?.replace(R.id.container, SoundsFragment())
+        fragmentTransaction?.addToBackStack(null)?.commit()
+    }
```

Comment:
should somehow set click on the buttomnavigation then it will navigate you

Reply:
I don't get this. This one is a function that changes the library's view to the list of sounds, when one of the category options are being pressed.
How is the bottom navigation related to this in any way?

Reply:
Oh it's my mistake. So the sound Fragment is not related to bottom navigation